openpype/hosts/nuke/api/template_loader.py

```diff
@@ -0,0 +1,534 @@
+            diff_y = max_y - min_y + 20
+            diff_x = abs(max_x - min_x - maxX + minX)
+            contained_nodes = considered_nodes
+
+        if diff_y > 0 or diff_x > 0:
+            for n in nodes:
+                refresh_node(n)
+                if n != node and n not in considered_nodes:
+
+                    if not isinstance(n, nuke.BackdropNode)\
+                            or isinstance(n, nuke.BackdropNode)\
+                            and not set(contained_nodes) <= set(n.getNodes()):
+                        if offset_y is None and n.xpos() >= min_x:
+                            n.setXpos(n.xpos() + diff_x)
+
+                        if n.ypos() >= min_y:
+                            n.setYpos(n.ypos() + diff_y)
+
+                    else:
+                        width = n.screenWidth()
+                        height = n.screenHeight()
+                        n.knob("bdwidth").setValue(width + diff_x)
+                        n.knob("bdheight").setValue(height + diff_y)
+
+                    refresh_node(n)
+
+    def imprint_inits(self):
+        """
+        add initial positions and dimensions to the attributes
+        """
+        for n in nuke.allNodes():
+            refresh_node(n)
+            imprint(n, {'x_init': n.xpos(), 'y_init': n.ypos()})
+            n.knob('x_init').setVisible(False)
+            n.knob('y_init').setVisible(False)
+            width = n.screenWidth()
+            height = n.screenHeight()
+            if 'bdwidth' in n.knobs().keys():
+                imprint(n, {'w_init': width, 'h_init': height})
+                n.knob('w_init').setVisible(False)
+                n.knob('h_init').setVisible(False)
+            refresh_node(n)
+
+    def imprint_siblings(self):
+        """
+        - add siblings names to placeholder attributes (nodes loaded with it)
+        - add Id to the attributes of all the other nodes
+        """
+
+        nodes_loaded = self.data['last_loaded']
+        d = {"id_rep": str(self.data['_id'])}
+
+        for n in nodes_loaded:
+            if "builder_type" in n.knobs().keys()\
+                    and ('is_placeholder' not in n.knobs().keys()
+                         or 'is_placeholder' in n.knobs().keys()
+                         and n.knob('is_placeholder').value()):
+
+                siblings = list(set(nodes_loaded) - set([n]))
+                siblings_name = get_names_from_nodes(siblings)
+                siblings = {"siblings": siblings_name}
+                imprint(n, siblings)
+
+            elif 'builder_type' not in n.knobs().keys():
+                # save the id of representation for all imported nodes
+                imprint(n, d)
+                n.knob('id_rep').setVisible(False)
+                refresh_node(n)
+
+    def set_loaded_connections(self):
+        """
+        set inputs and outputs of loaded nodes"""
+
+        node = self.data['node']
+        input, output = get_io(self.data['last_loaded'])
```

Comment:
There might be also some single unconnected nodes

```
Traceback (most recent call last):
  File "C:\CODE\__PYPE\OpenPype\openpype\hosts\nuke\api\pipeline.py", line 223, in <lambda>
    lambda: build_workfile_template()
  File "C:\CODE\__PYPE\OpenPype\openpype\lib\build_template.py", line 28, in build_workfile_template
    template_loader.populate_template()
  File "C:\CODE\__PYPE\OpenPype\openpype\hosts\nuke\api\template_loader.py", line 47, in populate_template
    super(NukeTemplateLoader, self).populate_template(ignored_ids)
  File "C:\CODE\__PYPE\OpenPype\openpype\lib\abstract_template_loader.py", line 254, in populate_template
    self.postload(placeholder)
  File "C:\CODE\__PYPE\OpenPype\openpype\lib\abstract_template_loader.py", line 306, in postload
    placeholder.clean()
  File "C:\CODE\__PYPE\OpenPype\openpype\hosts\nuke\api\template_loader.py", line 447, in clean
    self.set_loaded_connections()
  File "C:\CODE\__PYPE\OpenPype\openpype\hosts\nuke\api\template_loader.py", line 346, in set_loaded_connections
    input, output = get_io(self.data['last_loaded'])
  File "C:\CODE\__PYPE\OpenPype\openpype\hosts\nuke\api\lib.py", line 2628, in get_io
    raise Exception("No Input found")
Exception: No Input found
```

Reply:
but it might be perhaps different issue. I had connected the placeholder node to a dot and it still is having the same Traceback.. 
![image](https://user-images.githubusercontent.com/40640033/185127857-620e79eb-d943-4eaf-b6b4-00ee82363c83.png)